### DISCUSSION TITLE
Delta protocol version is too new for Athena DDL engine" Error When Querying Delta Tables

### DISCUSSION THREAD

**Discussion by marcoaanogueira** [delta-io/delta-rs#2939] [Q&A]
# Environment

**Delta-rs version**: 0.20.0

**Environment**:
- **Cloud provider**: AWS

***
# Bug

**What happened**:
 When trying to query Delta Lake tables via Athena, I received the following error:
Delta protocol version is too new for Athena DDL engine.
This error occurred even though the metadata shows a Delta protocol version that, according to the [documentation](https://docs.aws.amazon.com/athena/latest/ug/delta-lake-tables.html), should be compatible with Athena.

**What you expected to happen**:
Athena should be able to query the Delta Lake tables without encountering a protocol version incompatibility issue.

**How to reproduce it**:
Create a Delta Lake table in Athena using the following command:

```
CREATE EXTERNAL TABLE
<schema>.<table>
LOCATION '<delta table path>'
TBLPROPERTIES ('table_type' = 'DELTA')
```

**More details**:
<img width="939" alt="Captura de Tela 2024-10-10 às 12 13 02" src="https://github.com/user-attachments/assets/752d4491-6075-4c17-8576-0bf25dc5cf19">

**Comment by rtyler** [comment]
How was the table created? There may be some features used that pushed the protocol version too high for Athena SQL to use

**Comment by marcoaanogueira** [comment]
Sorry for closing it; I thought I should close it since the instruction in another issue was to open a support ticket with AWS.

I created the table using the write_deltalake function:
```
write_deltalake(
        table_or_uri=s3_destination,
        data=pa_source_delta_casted,
        mode="append",
        schema_mode="merge",
        storage_options=STORAGE_OPTIONS,
    )
```

**Comment by marcoaanogueira** [comment]
The version appears to be correct and compatible with what is stated in the AWS documentation.

<img width="804" alt="Captura de Tela 2024-10-10 às 14 29 03" src="https://github.com/user-attachments/assets/3553c588-9b4d-4ccc-830e-f080236306ff">

**Reply by rtyler** [reply]
I would not be surprised if this table was turned into minReaderVersion: 3,  minWriterVersion: 7 due to a timestamp without a timezone being in the data set used to create it. Thus it would be related to the tests I added recently in #2935 

Which documentation are you referring to? Athena is kind of a name used for Athena Spark and Athena SQL and those have different query engines and Delta support :scream:
